operator/src/main/java/com/github/streamshub/console/api/v1alpha1/spec/containers/ContainerTemplate.java

```diff
@@ -0,0 +1,49 @@
+package com.github.streamshub.console.api.v1alpha1.spec.containers;
+
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.ResourceRequirements;
+import io.sundr.builder.annotations.Buildable;
+
+@Buildable
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ContainerTemplate {
```

Comment:
Did you think of following the nomenclature  of https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-template-v1/#PodTemplateSpec?

In concrete terms, rename `ContainerTemplate` to `ContainerTemplateSpec` and have your ContainerSpec as a field within it.  At least that way you keep the door open to adding things like ContainerTemplateSpec.metadata.

Reply:
I think that makes sense. For an example, this means there is an added `spec` in the template, correct?

```yaml
spec:
  containers:
    api:
      spec:
        image: example.com/myorg/my-api-image:1.0
        resources:
          limits:
            cpu: 500m
            memory: 256Mi
        env:
        - name: SOME_ENVVAR
          value: "custom-value"
    ui:
      spec:
        image: example.com/myorg/my-ui-image:1.0
        resources:
          limits:
            cpu: 500m
            memory: 256Mi
        env:
        - name: SOME_ENVVAR_UI
          value: "custom-value"
```

Reply:
Suggested applied - nested the container spec information within the container template, leaving a nice opening for `metadata` in the future if necessary.